texar/modules/connectors/connectors.py

```diff
@@ -84,15 +84,36 @@ def _get_tensor_depth(x: torch.Tensor) -> int:
     """
     return int(np.prod(x.size()[1:]))
 
-def _mlp_transform(inputs: Inputs,
+
+def _sum_output_size(output_size: OutputSize) -> int:
+    r"""Return sum of all dim values in :attr:`output_size`
+
+    Args:
+        output_size: Can be an ``Integer``, a ``torch.Size``, or a (nested)
+            ``tuple`` of ``Integers`` or ``torch.Size``.
+    """
+    flat_output_size = nest.flatten(output_size)
+
+    if isinstance(flat_output_size[0], torch.Size):
+        size_list = [0] * len(flat_output_size)
+        for (i, shape) in enumerate(flat_output_size):
+            size_list[i] = np.prod([dim for dim in shape])
```

Comment:
This is a very un-Pythonic implementation. Readability could be greatly improved by writing:
```python
size_list = [np.prod(shape) for shape in flat_output_size]
```